xmodule/tests/test_item_bank.py

```diff
@@ -0,0 +1,498 @@
+        """
+        # Export self.item_bank to the virtual filesystem
+        export_fs = MemoryFS()
+        self.item_bank.runtime.export_fs = export_fs  # pylint: disable=protected-access
+        node = etree.Element("unknown_root")
+        self.item_bank.add_xml_to_node(node)
+
+        # Now, import the itembank using the same OLX as above, but with some XML comments in there.
+        # (Note: This will not work without exporting first, because it relies on the problem blocks OLX definitions
+        # being available on the filesystem)
+        olx_with_comments = (
+            '<!-- Comment -->\n'
+            '<itembank display_name="My Item Bank" max_count="1">\n'
+            '<!-- Comment -->\n'
+            '  <problem url_name="My_Item_0"/>\n'
+            '  <problem url_name="My_Item_1"/>\n'
+            '  <problem url_name="My_Item_2"/>\n'
+            '  <problem url_name="My_Item_3"/>\n'
+            '</itembank>\n'
+        )
+        olx_element = etree.fromstring(olx_with_comments)
+
+        # Import the olx.
+        runtime = TestImportSystem(load_error_blocks=True, course_id=self.item_bank.location.course_key)
+        runtime.resources_fs = export_fs
+        imported_item_bank = ItemBankBlock.parse_xml(olx_element, runtime, None)
+
+        # And make sure the result looks right.
+        self._verify_xblock_properties(imported_item_bank)
+
+    def _verify_xblock_properties(self, imported_item_bank):
+        """
+        Check the new XBlock has the same properties as the old one.
+        """
+        assert imported_item_bank.display_name == self.item_bank.display_name
+        assert imported_item_bank.max_count == self.item_bank.max_count
+        assert len(imported_item_bank.children) == len(self.item_bank.children)
+
+    def test_validation_of_matching_blocks(self):
+        """
+        Test that the validation method of LibraryContent blocks can warn
+        the user about problems with other settings (max_count and capa_type).
+        """
+        # Ensure we're starting wtih clean validation
+        assert self.item_bank.validate()
+
+        # Set max_count to higher value than exists in library
+        self.item_bank.max_count = 50
+        result = self.item_bank.validate()
+        assert not result
+        self._assert_has_only_N_matching_problems(result, 4)
+        assert len(self.item_bank.selected_children()) == 4
+
+        # Add some capa problems so we can check problem type validation messages
+        self.item_bank.max_count = 1
+        assert self.item_bank.validate()
+        assert len(self.item_bank.selected_children()) == 1
+
+        # -1 selects all blocks from the library.
+        self.item_bank.max_count = -1
+        assert self.item_bank.validate()
+        assert len(self.item_bank.selected_children()) == len(self.item_bank.children)
+
+        assert "@@TODO make more assertions about the validation messages"
```

Comment:
I'm actually removing most of the validation messages in my UI PR, except the one for when `max_count > num_children`. Though it would be good to add validation that max_count isn't some invalid value like `0` or `-2`.

Reply:
Done